librz/core/cmd_eval.c

```diff
@@ -498,142 +473,13 @@ RZ_IPI int rz_eval_color(void *data, const char *input) {
-					var++;
-			p = rz_sys_getenv(var);
-			if (p) {
-				rz_cons_println(p);
-				free(p);
-			} else {
-				char **e = rz_sys_get_environ();
-				while (e && *e) {
-					rz_cons_println(*e);
-					e++;
-				}
-			}
-		} else if (strlen(input) > 3) {
-			char *v, *k = strdup(input + 3);
-			if (!k)
-				break;
-			v = strchr(k, '=');
-			if (v) {
-				*v++ = 0;
-				rz_str_trim(k);
-				rz_str_trim(v);
-				rz_sys_setenv(k, v);
-			}
-			free(k);
-		}
-		return true;
-	case 'x': // exit
-		// XXX we need headers for the cmd_xxx files.
-		return rz_cmd_quit(data, "");
-	case 'j': // json
-		rz_config_list(core->config, NULL, 'j');
-		break;
-	case 'v': // verbose
-		rz_config_list(core->config, input + 1, 'v');
-		break;
-	case 'q': // quiet list of eval keys
-		rz_config_list(core->config, NULL, 'q');
-		break;
-	case 'c': // "ec"
-		rz_eval_color(core, input + 1);
-		break;
-	case 'e': // "ee"
-		if (input[1] == ' ') {
-			char *p;
-			const char *input2 = strchr(input + 2, ' ');
-			input2 = (input2) ? input2 + 1 : input + 2;
-			const char *val = rz_config_get(core->config, input2);
-			p = rz_core_editor(core, NULL, val);
-			if (p) {
-				rz_str_replace_char(p, '\n', ';');
-				rz_config_set(core->config, input2, p);
-			}
-		} else {
-			eprintf("Usage: ee varname # use $EDITOR to edit this config value\n");
-		}
-		break;
-	case '!': // "e!"
-		input = rz_str_trim_head_ro(input + 1);
-		if (!rz_config_toggle(core->config, input)) {
-			eprintf("rz_config: '%s' is not a boolean variable.\n", input);
-		}
-		break;
-	case 's': // "es"
-		rz_config_list(core->config, (input[1]) ? input + 1 : NULL, 's');
-		break;
-	case '-': // "e-"
-		rz_core_config_init(core);
-		//eprintf ("BUG: 'e-' command locks the eval hashtable. patches are welcome :)\n");
-		break;
-	case '*': // "e*"
-		rz_config_list(core->config, NULL, 1);
-		break;
-	case 'r': // "er"
-		if (input[1]) {
-			const char *key = input + ((input[1] == ' ') ? 2 : 1);
-			if (!rz_config_readonly(core->config, key)) {
-				eprintf("cannot find key '%s'\n", key);
-			}
-		} else {
-			eprintf("Usage: er [key]  # make an eval key PERMANENTLY read only\n");
-		}
-		break;
-	case ',': // "e."
-		rz_config_eval(core->config, input + 1, true);
-		break;
-	case '.': // "e "
-	case ' ': // "e "
-		if (rz_str_endswith(input, ".")) {
-			rz_config_list(core->config, input + 1, 0);
-		} else {
-			// XXX we cant do "e cmd.gprompt=dr=", because the '=' is a token, and quotes dont affect him
-			rz_config_eval(core->config, input + 1, false);
-		}
-		break;
-	}
-	return 0;
-}
-
 RZ_IPI RzCmdStatus rz_eval_getset_handler(RzCore *core, int argc, const char **argv) {
 	if (argc == 1) {
-		rz_config_list(core->config, NULL, 0);
+		RzCmdStateOutput state = { 0 };
+		rz_cmd_state_output_init(&state, RZ_OUTPUT_MODE_QUIET);
+		rz_core_config_print_all(core->config, NULL, &state);
+		rz_cmd_state_output_print(&state);
+		rz_cmd_state_output_fini(&state);
 		return RZ_CMD_STATUS_OK;
 	}
```

Comment:
actually this code is never reached, because `e` must have at least one argument... I guess you can remove it, because this is anyway covered by `elq`.